sklearn/preprocessing/imputation.py

```diff
@@ -299,13 +312,69 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
+    def _sparse_transform(self, X, missing_values, axis, valid_stats, valid_idx):
```

Comment:
I would rather `missing_values` and `axis` not be passed to this method, but retrieved from `self`. The current code makes it seem as if these might be different to the class attributes. And of course you're still accessing other parameters in this method, so their selection seems arbitrary.